components/service/experiments/src/main/java/mozilla/components/service/experiments/Experiments.kt

```diff
@@ -15,28 +21,91 @@ import mozilla.components.support.base.log.logger.Logger
  * @param valuesProvider provider for the device's values
  */
 @Suppress("TooManyFunctions")
-class Experiments(
-    private val source: ExperimentSource,
-    private val storage: ExperimentStorage,
-    valuesProvider: ValuesProvider = ValuesProvider()
-) {
+open class ExperimentsInternalAPI internal constructor() {
     @Volatile private var experimentsResult: ExperimentsSnapshot = ExperimentsSnapshot(listOf(), null)
     private var experimentsLoaded: Boolean = false
-    private val evaluator = ExperimentEvaluator(valuesProvider)
-    private val logger = Logger(LOG_TAG)
+    private var evaluator: ExperimentEvaluator = ExperimentEvaluator(ValuesProvider())
+    private lateinit var logger: Logger
+
+    private var source: ExperimentSource? = null
+    private var storage: ExperimentStorage? = null
+
+    /**
+     * Initialize the experiments library.
+     *
+     * This should only be initialized once by the application.
+     *
+     * @param applicationContext [Context] to access application features, such
+     * as shared preferences.
+     */
+    fun initialize(
+        applicationContext: Context
+    ) {
+        experimentsResult = ExperimentsSnapshot(listOf(), null)
+        experimentsLoaded = false
+        logger = Logger(LOG_TAG)
+
+        if (source == null) {
+            source = KintoExperimentSource(
+                EXPERIMENTS_BASE_URL,
+                EXPERIMENTS_BUCKET_NAME,
+                EXPERIMENTS_COLLECTION_NAME,
+                HttpURLConnectionClient()
+            )
+        }
+        if (storage == null) {
+            storage = FlatFileExperimentStorage(
+                // TODO: is this the right directory?
+                File(applicationContext.filesDir,
+                EXPERIMENTS_JSON_FILENAME)
+            )
+        }
+    }
+
+    /**
+     * Override the experiment source for testing.
+     *
+     * @param aSource the experiment source to use
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun overrideExperimentSource(aSource: ExperimentSource) {
+        source = aSource
+    }
+
+    /**
+     * Override the experiment storage for testing.
+     *
+     * @param aStorage the experiment storage to use
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun overrideExperimentStorage(aStorage: ExperimentStorage) {
```

Comment:
Instead of introducing these two functions, you could also just make `storage` an `internal` and `@VisibleForTesting` variable.